src/gui/sharemanager.cpp

```diff
@@ -278,6 +278,72 @@ void LinkShare::slotNameSet(const QJsonDocument &, const QVariant &value)
     emit nameSet();
 }
 
+UserGroupShare::UserGroupShare(AccountPtr account,
+    const QString &id,
+    const QString &owner,
+    const QString &ownerDisplayName,
+    const QString &path,
+    const ShareType shareType,
+    const Permissions permissions,
+    const QSharedPointer<Sharee> shareWith,
+    const QDate &expireDate,
+    const QString &note)
+    : Share(account, id, owner, ownerDisplayName, path, shareType, permissions, shareWith)
+    , _expireDate(expireDate)
+    , _note(note)
+{
+    Q_ASSERT(shareType == TypeUser || shareType == TypeGroup);
+    Q_ASSERT(shareWith);
+}
+
+void UserGroupShare::setNote(const QString &note)
+{
+    auto *job = new OcsShareJob(_account);
+    connect(job, &OcsShareJob::shareJobFinished, this, &UserGroupShare::slotNoteSet);
+    connect(job, &OcsJob::ocsError, this, &UserGroupShare::slotOcsError);
+    job->setNote(getId(), note);
+}
+
+QString UserGroupShare::getNote() const
+{
+    return _note;
+}
+
+void UserGroupShare::slotNoteSet(const QJsonDocument &, const QVariant &note)
+{
+    _note = note.toString();
+    emit noteSet();
+}
+
+QDate UserGroupShare::getExpireDate() const
+{
+    return _expireDate;
+}
+
+void UserGroupShare::setExpireDate(const QDate &date)
+{
+    auto *job = new OcsShareJob(_account);
+    connect(job, &OcsShareJob::shareJobFinished, this, &UserGroupShare::slotExpireDateSet);
+    connect(job, &OcsJob::ocsError, this, &UserGroupShare::slotOcsError);
+    job->setExpireDate(getId(), date);
+}
+
+void UserGroupShare::slotExpireDateSet(const QJsonDocument &reply, const QVariant &value)
+{
+    auto data = reply.object().value("ocs").toObject().value("data").toObject();
+
+    /*
+     * If the reply provides a data back (more REST style)
+     * they use this date.
+     */
+    if (data.value("expiration").isString()) {
+        _expireDate = QDate::fromString(data.value("expiration").toString(), "yyyy-MM-dd 00:00:00");
+    } else {
+        _expireDate = value.toDate();
+    }
+    emit expireDateSet();
```

Comment:
@FlexW I'd say, you may also want to check `if (_expireDate.isValid()` with `qCWarning` and `Q_ASSERT `before emit. Just in case.

Reply:
Yes makes sense:)

Reply:
Actually I found out that the logic depends on invalid dates. If one wants to delete the expire date on the server a invalid date get send.